ext/v8/integer.h

```diff
@@ -0,0 +1,47 @@
+// -*- mode: c++ -*-
+#ifndef RR_INTEGER_H
+#define RR_INTEGER_H
+
+namespace rr {
+  class Integer : public Ref<v8::Integer> {
+  public:
+    Integer(v8::Isolate* isolate, v8::Handle<v8::Integer> integer) :
+      Ref<v8::Integer>(isolate, integer) {}
+    Integer(VALUE self) :
+      Ref<v8::Integer>(self) {}
+
+    static VALUE New(VALUE self, VALUE r_isolate, VALUE value) {
+      Isolate isolate(r_isolate);
+      Locker lock(isolate);
+
+      v8::Local<v8::Integer> i = v8::Integer::New(isolate, NUM2INT(value));
+      return Value::handleToRubyObject(isolate, i);
```

Comment:
So the main reason is because I also can't really wrap my head around how Integers work either. I believe that when you call `v8::Integer::New`, and it will either return `Int32, Uint32, or just Integer`. Since I'm not really sure what's going internally, I opted for making the call to `New`, and then downcasting via the call to `handleToRubyObject`

Turns out `v8::Handle<T>` is an alias for `v8::Loca<T>` https://github.com/v8/v8-git-mirror/blob/4.5.90/include/v8.h#L333-L335

It used to be the superclass, but now it's the same thing.